File: src/leetcode/2026/04/Solution867.java
public class Solution867 {
    public int[][] transpose(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        int[][] transposed = new int[n][m];
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                transposed[j][i] = matrix[i][j];
            }
        }
        return transposed;
    }

    public static void main(String[] args) {
        Solution867 solution = new Solution867();

        int[][] matrix1 = { { 1, 2, 3 }, { 4, 5, 6 }, { 7, 8, 9 } };
        int[][] transposed1 = solution.transpose(matrix1);
        printMatrix(transposed1); // Output: [[1,4,7],[2,5,8],[3,6,9]]

        int[][] matrix2 = { { 1, 2 }, { 3, 4 }, { 5, 6 } };
        int[][] transposed2 = solution.transpose(matrix2);
        printMatrix(transposed2); // Output: [[1,3,5],[2,4,6]]

        int[][] matrix3 = { { 1, 2, 3 }, { 4, 5, 6 } };
        int[][] transposed3 = solution.transpose(matrix3);
        printMatrix(transposed3); // Output: [[1,4],[2,5],[3,6]]
    }

    private static void printMatrix(int[][] matrix) {
        for (int[] row : matrix) {
            System.out.print("[");
            for (int i = 0; i < row.length; i++) {
                System.out.print(row[i]);
                if (i < row.length - 1) {
                    System.out.print(",");
                }
            }
            System.out.println("]");
        }
    }
}

File: src/leetcode/2026/04/Solution867.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution867 {
public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) {
        int m = matrix.size();
        int n = matrix[0].size();
        vector<vector<int>> transposed(n, vector<int>(m));
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                transposed[j][i] = matrix[i][j];
            }
        }
        return transposed;
    }
};

int main() {
    Solution867 solution;

    vector<vector<int>> matrix1 = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
    vector<vector<int>> result1 = solution.transpose(matrix1);
    for (const auto& row : result1) {
        for (int val : row) {
            cout << val << " ";
        }
        cout << endl;
    }

    vector<vector<int>> matrix2 = {{1, 2, 3}, {4, 5, 6}};
    vector<vector<int>> result2 = solution.transpose(matrix2);
    for (const auto& row : result2) {
        for (int val : row) {
            cout << val << " ";
        }
        cout << endl;
    }

    return 0;
}